log4j-api/src/main/java/org/apache/logging/log4j/ResourceLogger.java

```diff
@@ -0,0 +1,276 @@
+        return logger.isEnabled(level, marker, message, p0, p1, p2);
+    }
+
+    @Override
+    public boolean isEnabled(Level level, Marker marker, String message, Object p0, Object p1, Object p2, Object p3) {
+        return logger.isEnabled(level, marker, message, p0, p1, p2, p3);
+    }
+
+    @Override
+    public boolean isEnabled(
+            Level level, Marker marker, String message, Object p0, Object p1, Object p2, Object p3, Object p4) {
+        return logger.isEnabled(level, marker, message, p0, p1, p2, p3, p4);
+    }
+
+    @Override
+    public boolean isEnabled(
+            Level level,
+            Marker marker,
+            String message,
+            Object p0,
+            Object p1,
+            Object p2,
+            Object p3,
+            Object p4,
+            Object p5) {
+        return logger.isEnabled(level, marker, message, p0, p1, p2, p3, p4, p5);
+    }
+
+    @Override
+    public boolean isEnabled(
+            Level level,
+            Marker marker,
+            String message,
+            Object p0,
+            Object p1,
+            Object p2,
+            Object p3,
+            Object p4,
+            Object p5,
+            Object p6) {
+        return logger.isEnabled(level, marker, message, p0, p1, p2, p3, p4, p5, p6);
+    }
+
+    @Override
+    public boolean isEnabled(
+            Level level,
+            Marker marker,
+            String message,
+            Object p0,
+            Object p1,
+            Object p2,
+            Object p3,
+            Object p4,
+            Object p5,
+            Object p6,
+            Object p7) {
+        return logger.isEnabled(level, marker, message, p0, p1, p2, p3, p4, p5, p6, p7);
+    }
+
+    @Override
+    public boolean isEnabled(
+            Level level,
+            Marker marker,
+            String message,
+            Object p0,
+            Object p1,
+            Object p2,
+            Object p3,
+            Object p4,
+            Object p5,
+            Object p6,
+            Object p7,
+            Object p8) {
+        return logger.isEnabled(level, marker, message, p0, p1, p2, p3, p4, p5, p6, p7, p8);
+    }
+
+    @Override
+    public boolean isEnabled(
+            Level level,
+            Marker marker,
+            String message,
+            Object p0,
+            Object p1,
+            Object p2,
+            Object p3,
+            Object p4,
+            Object p5,
+            Object p6,
+            Object p7,
+            Object p8,
+            Object p9) {
+        return logger.isEnabled(level, marker, message, p0, p1, p2, p3, p4, p5, p6, p7, p8, p9);
+    }
+
+    @Override
+    public void logMessage(String fqcn, Level level, Marker marker, Message message, Throwable t) {
+        logger.logMessage(fqcn, level, marker, message, t);
+    }
+
+    /**
+     * Constructs a ResourceLogger.
+     */
+    public static final class ResourceLoggerBuilder {
```

Comment:
Personally I would prefer to just have a `ResourceLoggerBuilder` (or `LoggerBuilder`) **interface** and access it through a new `Logger#newDetachedLoggerBuilder` method.

Creating "detached" loggers might be used for other purposes than adding key/value pairs to a message. For example someone might want to use a different `MessageFactory`.